src/citrine/resources/predictor_evaluation_execution.py

```diff
@@ -0,0 +1,168 @@
+    workflow_id = properties.UUID('workflow_id', serializable=False)
+    predictor_id = properties.UUID('predictor_id', serializable=False)
+    status = properties.Optional(properties.String(), 'status', serializable=False)
+    status_info = properties.Optional(
+        properties.List(properties.String()),
+        'status_info',
+        serializable=False
+    )
+    experimental = properties.Boolean("experimental", serializable=False, default=True)
+    experimental_reasons = properties.Optional(
+        properties.List(properties.String()),
+        'experimental_reasons',
+        serializable=False
+    )
+
+    def __init__(self):
+        self.project_id: Optional[UUID] = None
+        self.session: Optional[Session] = None
+
+    def __str__(self):
+        return '<PredictorEvaluationExecution {!r}>'.format(str(self.uid))
+
+    def _path(self):
+        return '/projects/{project_id}/predictor-evaluation-executions/{execution_id}' \
+            .format(project_id=self.project_id,
+                    workflow_id=self.workflow_id,
+                    execution_id=self.uid)
+
+    @lru_cache()
+    def results(self, evaluator_name: str) -> PredictorEvaluationResult:
+        """
+        Get a specific evaluation result by the name of the evaluator that produced it.
+
+        Parameters
+        ----------
+        evaluator_name: str
+            Name of the evaluator for which to get the results
+
+        Returns
+        -------
+        The evaluation result from the evaluator with the given name
+
+        """
+        params = {"evaluator_name": evaluator_name}
+        resource = self.session.get_resource(self._path() + "/results", params=params)
+        return PredictorEvaluationResult.build(resource)
+
+    def __getitem__(self, item):
+        if isinstance(item, str):
+            return self.results(item)
+        else:
+            raise TypeError("Results are accessed by string names")
+
+    def __iter__(self):
+        return iter(self.response_names)
+
+
+class PredictorEvaluationExecutionCollection(Collection["PredictorEvaluationExecution"]):
+    """[ALPHA] A collection of PredictorEvaluationExecutions."""
+
+    _path_template = '/projects/{project_id}/predictor-evaluation-executions'  # noqa
+    _individual_key = None
+    _collection_key = 'response'
+    _resource = PredictorEvaluationExecution
+
+    def __init__(self, *,
+                 project_id: UUID,
+                 session: Session,
+                 workflow_id: Optional[UUID] = None,
+                 ):
+        self.project_id: UUID = project_id
+        self.session: Session = session
+        self.workflow_id: Optional[UUID] = workflow_id
+
+    def build(self, data: dict) -> PredictorEvaluationExecution:
+        """Build an individual PredictorEvaluationExecution."""
+        execution = PredictorEvaluationExecution.build(data)
+        execution.session = self.session
+        execution.project_id = self.project_id
```

Comment:
does this need to set the `workflow_id` too?

Reply:
No, since that's returned by the API.